pkg/cmd/step/create/step_create_install_secrets.go

```diff
@@ -0,0 +1,142 @@
+package create
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/jenkins-x/jx/pkg/cmd/helper"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/templates"
+	"github.com/jenkins-x/jx/pkg/helm"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/jenkins-x/jx/pkg/util"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/engine"
+)
+
+const (
+	// SecretsTemplateFileName contains the template used to generate the secrets.yaml file which can be safely checked into git
+	SecretsTemplateFileName = "template-secrets.yaml"
+
+	// SecretsParametersFileName the secret parameter files which are never checked into git and may come from, say, Vault
+	SecretsParametersFileName = "secrets-parameters.yaml"
```

Comment:
Can we use in this file the vault's URI based secrets lookup? The purpose of URI based secrets lookup was exactly to avoid committing secrets in git and have a dynamic mechanism to resolve thme. The secrets are resolved and interpolated before applying the charts. 

https://github.com/jenkins-x/environment-tekton-mole-dev/blob/783a8f6178c050c0b96c61016d1581bd0bc7a3a0/env/values.yaml#L92

I think the existing vault URI based lookup can be extended for k8s secrets and local secrets as follows: 

Vault:

``` 
vault:gitOps/jenkins-x/environment-tekton-mole-dev/feature-flag-api-key:token-passthrough
````

k8s secrets:
```
 k8s:<namespace>/<secret-name>:<secret-value>
```
Local:
```
```

```
 local:<yaml-file-path>/<secret-secret-path-in-yaml>:<secret-value>
```

Reply:
yeah - I did wonder about using a function in the template for this kinda thing - so `{{ .Vault.gitOps/jenkins-x/environment-tekton-mole-dev/feature-flag-api-key:token-passthrough }}` or something like that?

Reply:
Agree with @ccojocar.

* anything where there is a risk you might commit the actual secrets, we should avoid like the plauge
* the secrets plugin is odd and confusing, I think we should try to not use it at all

The URL approach is much better in my opinion and I think we can use the scheme Cosmin proposes to generalise it.

I'm not sure why we would need to use a function - is this when you are using non tillerless helm?